hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1874,7 +1874,7 @@ public static HoodieData<HoodieRecord> readSecondaryKeysFromBaseFiles(HoodieEngi
       List<String> logFilePaths = new ArrayList<>();
       baseAndLogFiles.getValue().forEach(logFile -> logFilePaths.add(basePath + StoragePath.SEPARATOR + partition + StoragePath.SEPARATOR + logFile));
       String filePath = baseAndLogFiles.getKey();
-      Option<StoragePath> dataFilePath = filePath.isEmpty() ? Option.empty() : Option.of(filePath(basePath, partition, filePath));
+      Option<StoragePath> dataFilePath = filePath.isEmpty() ? Option.empty() : Option.of(FSUtils.constructAbsolutePath(basePath, filePath));
```

Comment:
comment on L1875. Do we have tests for non-partitioned table for functional and sec index. 
```
baseAndLogFiles.getValue().forEach(logFile -> logFilePaths.add(basePath + StoragePath.SEPARATOR + partition + StoragePath.SEPARATOR + logFile));
```

This could create a path w/ 2 separators (/) for non-partitioned table.

Reply:
Additional separator will be ignored. Added non-partition test. I have made all tests parameterized over tableType and (non)partition.